drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3628,29 +3643,41 @@ void rswitch_add_ipv4_forward_all_types(struct l3_ipv4_fwd_param *param, struct
 	}
 
 	param_list[0].param->frame_type = LTHSLP0v4OTHER;
+	param_list[1].param->frame_type = LTHSLP0v4UDP;
+	param_list[2].param->frame_type = LTHSLP0v4TCP;
+
+	mutex_lock(&priv->ipv4_forward_lock);
+
+	list_add(&param_list[0].list, &routing_list->param_list);
```

Comment:
If you move list_add section before hw setup, you have to add removing them in case of any hw error too.